.cross_sync/transformers.py

```diff
@@ -0,0 +1,321 @@
+        return self.visit_FunctionDef(node)
+
+    def visit_FunctionDef(self, node):
+        """
+        Replace function docstrings
+        """
+        docstring = ast.get_docstring(node)
+        if docstring and isinstance(node.body[0], ast.Expr) and isinstance(
+            node.body[0].value, ast.Str
+        ):
+            for key_word, replacement in self.replacements.items():
+                docstring = docstring.replace(key_word, replacement)
+            node.body[0].value.s = docstring
+        return self.generic_visit(node)
+
+    def visit_Constant(self, node):
+        """Replace string type annotations"""
+        node.s = self.replacements.get(node.s, node.s)
+        return node
+
+
+class AsyncToSync(ast.NodeTransformer):
+    """
+    Replaces or strips all async keywords from a given AST
+    """
+    def visit_Await(self, node):
+        """
+        Strips await keyword
+        """
+        return self.visit(node.value)
+
+    def visit_AsyncFor(self, node):
+        """
+        Replaces `async for` with `for`
+        """
+        return ast.copy_location(
+            ast.For(
+                self.visit(node.target),
+                self.visit(node.iter),
+                [self.visit(stmt) for stmt in node.body],
+                [self.visit(stmt) for stmt in node.orelse],
+            ),
+            node,
+        )
+
+    def visit_AsyncWith(self, node):
+        """
+        Replaces `async with` with `with`
+        """
+        return ast.copy_location(
+            ast.With(
+                [self.visit(item) for item in node.items],
+                [self.visit(stmt) for stmt in node.body],
+            ),
+            node,
+        )
+
+    def visit_AsyncFunctionDef(self, node):
+        """
+        Replaces `async def` with `def`
+        """
+        return ast.copy_location(
+            ast.FunctionDef(
+                node.name,
+                self.visit(node.args),
+                [self.visit(stmt) for stmt in node.body],
+                [self.visit(decorator) for decorator in node.decorator_list],
+                node.returns and self.visit(node.returns),
+            ),
+            node,
+        )
+
+    def visit_ListComp(self, node):
+        """
+        Replaces `async for` with `for` in list comprehensions
+        """
+        for generator in node.generators:
+            generator.is_async = False
+        return self.generic_visit(node)
+
+class RmAioFunctions(ast.NodeTransformer):
+    """
+    Visits all calls marked with CrossSync.rm_aio, and removes asyncio keywords
+    """
+
+    def __init__(self):
+        self.to_sync = AsyncToSync()
+
+    def visit_Call(self, node):
+        if isinstance(node.func, ast.Attribute) and isinstance(node.func.value, ast.Name) and \
+        node.func.attr == "rm_aio" and "CrossSync" in node.func.value.id:
+            return self.visit(self.to_sync.visit(node.args[0]))
+        return self.generic_visit(node)
```

Comment:
`.visit()` would call `RmAioFunctions.visit_Call(node)` again for the node, and `generic_visit` [bypasses the visit_Call overload](https://github.com/python/cpython/blob/500f5338a8fe13719478589333fcd296e8e8eb02/Lib/ast.py#L517), and just continues processing its children instead

We need to use generic_visit here to avoid infinite recursion, basically saying to ignore this node and keep going